test/suites/integration/launchtemplates_test.go

```diff
@@ -61,6 +63,46 @@ var _ = Describe("LaunchTemplates", func() {
 
 		ExpectInstance(pod.Spec.NodeName).To(HaveField("ImageId", HaveValue(Equal(customAMI))))
 	})
+	It("should merge UserData contents for AL2 AMIFamily", func() {
+		content, _ := ioutil.ReadFile("testdata/al2_userdata_input.golden")
+		provider := test.AWSNodeTemplate(v1alpha1.AWSNodeTemplateSpec{AWS: awsv1alpha1.AWS{
+			SecurityGroupSelector: map[string]string{"karpenter.sh/discovery": env.ClusterName},
+			SubnetSelector:        map[string]string{"karpenter.sh/discovery": env.ClusterName},
+			AMIFamily:             &awsv1alpha1.AMIFamilyAL2,
+		},
+			UserData: aws.String(string(content)),
+		})
+		provisioner := test.Provisioner(test.ProvisionerOptions{ProviderRef: &v1alpha5.ProviderRef{Name: provider.Name}})
+		pod := test.Pod()
+
+		env.ExpectCreated(pod, provider, provisioner)
+		env.EventuallyExpectHealthy(pod)
+		env.ExpectCreatedNodeCount("==", 1)
+
+		actualUserData, _ := base64.StdEncoding.DecodeString(*getInstanceAttribute(pod.Spec.NodeName, "userData").UserData.Value)
```

Comment:
Needs an Expect(err).ToNot(HaveOccurred) here.

Reply:
I don't think we need to be validating EC2's API response since they've got a contract that their UserData is always Base64 encoded